builder/azure/arm/template_factory.go

```diff
@@ -72,6 +72,18 @@ func GetVirtualMachineDeployment(config *Config) (*resources.Deployment, error)
 			config.ImageVersion)
 
 		builder.SetManagedMarketplaceImage(config.Location, config.ImagePublisher, config.ImageOffer, config.ImageSku, config.ImageVersion, imageID, config.managedImageStorageAccountType)
+	} else if config.SharedGallerySubscription != "" && config.SharedGalleryResourceGroup != "" && config.SharedGalleryName != "" && config.SharedGalleryImageName != "" {
```

Comment:
Do you need all of these checks, or is it sufficient to use just SharedGallerySubscription?  If you've gotten this far the configuration checks had to all pass.